src/gamepad/mappings-manager.vala

```diff
@@ -0,0 +1,99 @@
+// This file is part of GNOME Games. License: GPLv3
+
+/**
+ * This class gives methods to set/update the mappings
+ *
+ * The client interfaces with this class primarily
+ */
+public class LibGamepad.MappingsManager {
+	private static HashTable<string, string> names;
+	private static HashTable<string, string> mappings;
+
+	private static bool? inited;
```

Comment:
This was due to the static problem but as we are switching to singleton this can not be a problem anymore (I was unable to assign a static to some value)